File: Publisher_Subscriber/ex_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "a_node_name_2");
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::String>("/a_topic_name",10);

	ros::Rate rate(3);

	while(ros::ok())
	{
		std_msgs::String msg;
		msg.data = "Hi, this is a publisher";
		pub.publish(msg);
		rate.sleep();
	}
}

File: Publisher_Subscriber/ex_publisher.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Author: hbasriaykol

import rospy
from std_msgs.msg import String


def create_publisher():
    """
     This function is a publisher node.
    """
    pub = rospy.Publisher("/a_topic_name", String, queue_size=10)
    # publisher(topic_name, msgs_type, queue_size)
    rate = rospy.Rate(2)

    while not rospy.is_shutdown():
        msg = String()
        msg.data = "Hi, this is a publisher."
        pub.publish(msg)
        rate.sleep()


if __name__ == '__main__':
    rospy.init_node('a_node_name_2')
    create_publisher()
    rospy.loginfo("Node was stopped.")
